runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/streaming/ComputationStateCacheTest.java

```diff
@@ -0,0 +1,249 @@
+
+  private static Work createWork(long workToken, long cacheToken) {
+
+    return Work.create(
+        Windmill.WorkItem.newBuilder()
+            .setKey(ByteString.copyFromUtf8(""))
+            .setShardingKey(1)
+            .setWorkToken(workToken)
+            .setCacheToken(cacheToken)
+            .build(),
+        Instant::now,
+        Collections.emptyList(),
+        unused -> {});
+  }
+
+  @Before
+  public void setUp() {
+    computationStateCache =
+        ComputationStateCache.create(
+            configFetcher, workExecutor, ignored -> stateCache, IdGenerators.decrementingLongs());
+  }
+
+  @Test
+  public void testGet_computationStateCached() {
+    String computationId = "computationId";
+    MapTask mapTask = new MapTask().setStageName("stageName").setSystemName("systemName");
+    Map<String, String> userTransformToStateFamilyName =
+        ImmutableMap.of("userTransformName", "stateFamilyName");
+    ComputationConfig computationConfig =
+        ComputationConfig.create(mapTask, userTransformToStateFamilyName, ImmutableMap.of());
+    when(configFetcher.fetchConfig(eq(computationId))).thenReturn(Optional.of(computationConfig));
+    computationStateCache.get(computationId);
+    Optional<ComputationState> computationState = computationStateCache.get(computationId);
+    assertTrue(computationState.isPresent());
+    assertThat(computationState.get().getComputationId()).isEqualTo(computationId);
+    assertThat(computationState.get().getMapTask()).isEqualTo(mapTask);
+    assertThat(computationState.get().getTransformUserNameToStateFamily())
+        .isEqualTo(userTransformToStateFamilyName);
+  }
+
+  @Test
+  public void testGet_computationStateNotCached() {
+    String computationId = "computationId";
+    MapTask mapTask = new MapTask().setStageName("stageName").setSystemName("systemName");
+    Map<String, String> userTransformToStateFamilyName =
+        ImmutableMap.of("userTransformName", "stateFamilyName");
+    ComputationConfig computationConfig =
+        ComputationConfig.create(mapTask, userTransformToStateFamilyName, ImmutableMap.of());
+    when(configFetcher.fetchConfig(eq(computationId))).thenReturn(Optional.of(computationConfig));
+    Optional<ComputationState> computationState = computationStateCache.get(computationId);
+    assertTrue(computationState.isPresent());
+    assertThat(computationState.get().getComputationId()).isEqualTo(computationId);
+    assertThat(computationState.get().getMapTask()).isEqualTo(mapTask);
+    assertThat(computationState.get().getTransformUserNameToStateFamily())
+        .isEqualTo(userTransformToStateFamilyName);
+  }
+
+  @Test
+  public void testGet_computationStateNotCachedOrFetchable() {
+    String computationId = "computationId";
+    when(configFetcher.fetchConfig(eq(computationId))).thenReturn(Optional.empty());
+    Optional<ComputationState> computationState = computationStateCache.get(computationId);
+    assertFalse(computationState.isPresent());
```

Comment:
call get again and verify the fetch occurs

Reply:
done